src/cli/mod.ts

```diff
@@ -96,6 +102,81 @@ export function dzx() {
       }),
     );
 
+  async function moduleFromMarkdown(md: string) {
+    let mdRealPath;
+    let mdContent;
+
+    if (
+      md.startsWith("http://") ||
+      md.startsWith("https://")
+    ) {
+      mdContent = await fetch(md).then((r) => r.text());
+      mdRealPath = await Deno.makeTempFile({
+        prefix: "dzx_",
+        suffix: `_source.md`,
+      });
+
+      await Deno.writeTextFile(mdRealPath, mdContent);
+
+      console.log($.dim(`Markdown source saved to ${mdRealPath}`));
+    }
+
+    mdRealPath ??= await Deno.realPath(md);
+    mdContent ??= await Deno.readTextFile(mdRealPath);
+    const mdFileURL = path.toFileUrl(await Deno.realPath(mdRealPath));
+    const mdTokens = tokens(mdContent);
+
+    const validLangs = ["js", "javascript", "ts", "typescript"];
+    const codeContent: string[] = [];
+
+    mdTokens.forEach((token, idx) => {
+      if (
+        token.type === "start" && token.tag === "codeBlock" &&
+        token.kind === "fenced" &&
+        validLangs
+          .includes(token.language)
+      ) {
+        let cursor = idx + 1;
+
+        while (mdTokens.at(cursor)?.type === "text") {
+          const code = mdTokens.at(cursor);
+
+          // Silly typescript, can't narrow this down based on
+          // the while loop condition...ah well
+          if (code?.type === "text") {
+            codeContent.push(code.content);
+          }
+
+          cursor++;
+        }
+      }
+    });
+
+    const tmp = await Deno.makeTempFile({
+      prefix: "dzx_",
+      suffix: `_module.ts`,
+    });
+
+    console.log($.dim(`Markdown module saved to ${tmp}\n`));
+
+    const finalCode = codeContent.join("").replaceAll(
+      "import.meta.url",
+      `\"${mdFileURL}\"`,
+    );
+
+    await Deno.writeTextFile(
+      tmp,
+      `#!/usr/bin/env dzx
+       /// <reference path="https://deno.land/x/dzx@${VERSION}/types.d.ts" />
+       // deno-lint-ignore-file
+
+       ${finalCode}
+      `,
+    );
+
+    return tmp;
```

Comment:
i think we need to add the file protocol here:
```suggestion
    return "file://" + tmp;
```
otherwise the dynamic import won't work if you install dzx from an url.
```console
Markdown module saved to /var/folders/1_/yy85dgv13_sdrsqg5wv43zyr0000gn/T/dzx_f458570e_module.ts

Download https://raw.githubusercontent.com/var/folders/1_/yy85dgv13_sdrsqg5wv43zyr0000gn/T/dzx_f458570e_module.ts
error: Uncaught (in promise) TypeError: Import 'https://raw.githubusercontent.com/var/folders/1_/yy85dgv13_sdrsqg5wv43zyr0000gn/T/dzx_f458570e_module.ts' failed: 404 Not Found
            await import($.mainModule);
```

Reply:
I'm fairly certain that the error you see here is because by default, when you visit a bad url in the `raw.githubusercontent.com` domain, it gives you back a valid text response that says literally "404 Not Found".

It looks like you might have saved that "404 Not Found" response manually to a file that doesn't have a `.md` extension, so this hit the typical code path, and thus it tried to parse it as a regular script.

I'm thinking that your stack trace above is this actually working correctly, but regardless, I did push a quick change to call your original `addProtocol` function anyway, just in case :)

Reply:
it was because of the missing file protocol, becuase if you don't specify a protocol it will always try to import the file from the host of `import.meta.url` of your main module. And if you install dzx from your github branch with `deno install --reload --allow-all --unstable -f https://raw.githubusercontent.com/andrewbrey/deno-dzx/feat/markdown-scripts/dzx.ts` it trys to load all relative and absolute imports from `https://raw.githubusercontent.com`.
But your last commit fixed it, now it works 👍